### PR TITLE
🐛 Fixes: Bugs observed during Mithril1 prerelease

### PR DESCRIPTION
**Pull request by mrnicegyu11**
## What do these changes do?
- 🐛 Fix bug: Deployment agent v0.11.1 complains/spams in mattermost when commits of watched git-repos are re-tagged (fix via https://stackoverflow.com/a/58438257 )
- 🐛 Fix bug: Deployment agent needs to die when init fails due to missing tag-sync, since the GitUrlWatcher object will not be initialized and not be ready for polling. A failure and full restart of the deployment agent is (without further refactoring) the more graceful option out of this

See as well: https://github.com/ITISFoundation/osparc-issues/issues/896#issuecomment-1468326225

### CODE REVIEW

File: src/simcore_service_deployment_agent/git_url_watcher.py [hunk 1]
```diff
@@ -178,19 +178,19 @@ async def _git_clean_repo(directory: str):
 
 async def _git_checkout_files(directory: str, paths: list[Path], tag: Optional[str]):
     if not tag:
-        tag = "HEAD"
-    cmd = ["git", "checkout", tag] + [f"{path}" for path in paths]
+        tag: Literal["HEAD"] = "HEAD"
```
Comment (pcrespov) [inline src/simcore_service_deployment_agent/git_url_watcher.py:181]: TIP: re-annotating a variable is not a very good practice.  An alternative implementation
``python
async def _git_checkout_files(directory: str, paths: list[Path], tag: Optional[str]):tag = tag or "HEAD"
    tag = tag or "HEAD"
    # or 
   tag = "HEAD" if tag is None else tag
``

**Review by pcrespov — APPROVED** [review]

File: src/simcore_service_deployment_agent/git_url_watcher.py (hunk 1 above)
Comment (mrnicegyu11) [reply src/simcore_service_deployment_agent/git_url_watcher.py:181]: will change this, thx

File: src/simcore_service_deployment_agent/auto_deploy_task.py [hunk 2]
```diff
@@ -433,9 +433,11 @@ async def auto_deploy(app: web.Application):
         app["state"][TASK_NAME] = State.STOPPED
         return
     except TagSyncErrorException:
-        log.warning(
-            "Problem  while initializing deployment: Tag-Sync specified but latest tags did not match. Continuing polling..."
+        log.error(
+            "Problem while initializing deployment: Tag-Sync specified but latest tags did not match. Continuing polling..."
```
Comment (sanderegg) [inline src/simcore_service_deployment_agent/auto_deploy_task.py:437]: I guess that if the application now crashes on start, the message shall be changed...

**Review by sanderegg — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]
the commit list is a bit weird with regard to the number of effective changes...

File: src/simcore_service_deployment_agent/auto_deploy_task.py (hunk 2 above)
Comment (mrnicegyu11) [reply src/simcore_service_deployment_agent/auto_deploy_task.py:437]: I will address this

**Pull request merged**